projects/HVAC/vent_control/vent.ino

```diff
@@ -0,0 +1,260 @@
+/*
+
+motor: 28BYJ48-5V
+  steps: 8
+  stride angle: 5.625/64
+
+  loops to 1 revolution: 360/(8 * 5.625/64) = 512
+
+            | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
+  4 orange  | x | x |   |   |   |   |   | x |
+  ----------|---|---|---|---|---|---|---|---|
+  3 yellow  |   | x | x | x |   |   |   |   |
+  ----------|---|---|---|---|---|---|---|---|
+  2 pink    |   |   |   | x | x | x |   |   |
+  ----------|---|---|---|---|---|---|---|---|
+  1 blue    |   |   |   |   |   | x | x | x |
+  ----------|---|---|---|---|---|---|---|---|
+
+Example:
+
+@VENT01>set degrees: 90
+@VENT01>set percents: 25
+
+*/
+
+#include <SyrotaAutomation1.h>
+
+void motor_step_cw();
+void motor_step_ccw();
+bool motor_zero();
+void motor_rotate_degrees(int degrees);
+void motor_rotate_percents(int percents);
+void motor_rotate_to(int position);
+
+void set_error(char *message);
+
+// Handles:
+//   * set degrees: -360..360
+//   * set percents: -100..100
+void process_command();
+
+int motor_position = 0;
+// steps per rotation
+const int motor_steps_per_rotation = 512;
+const float motor_steps_per_degree = motor_steps_per_rotation / 360.0;
+
+const int motor_sensor_pin = 2;
+const int blink_pin = 13;
+
+const int serial_pin = 3;
+SyrotaAutomation net = SyrotaAutomation(serial_pin);
+const char* net_address = "VENT01";
+
+
+void setup() {
+  // Serial
+  Serial.begin(9600);
+
+  // Error light
+  pinMode(blink_pin, OUTPUT);
+  digitalWrite(blink_pin, LOW);
+
+  // Motor pins
+  pinMode(8,  OUTPUT);
+  pinMode(9,  OUTPUT);
+  pinMode(10, OUTPUT);
+  pinMode(11, OUTPUT);
+
+  pinMode(motor_sensor_pin, INPUT);
```

Comment:
This needs to be INPUT_PULLUP, as pin is unconnected when button is not pressed (meaning it will read randomly 0 or 1 depending on what happens to nearby pins, interference, etc). When button is pressed, it is connected to GND.